aiokafka/protocol/types.py

```diff
@@ -253,36 +300,36 @@ def encode(cls, value):
         return ret
 
 
-class VarInt32(AbstractType):
+class VarInt32(AbstractType[int]):
     @classmethod
-    def decode(cls, data):
+    def decode(cls, data: BytesIO) -> int:
         value = UnsignedVarInt32.decode(data)
         return (value >> 1) ^ -(value & 1)
 
     @classmethod
-    def encode(cls, value):
+    def encode(cls, value: int) -> bytes:
         # bring it in line with the java binary repr
         value &= 0xFFFFFFFF
         return UnsignedVarInt32.encode((value << 1) ^ (value >> 31))
 
 
-class VarInt64(AbstractType):
+class VarInt64(AbstractType[int]):
     @classmethod
-    def decode(cls, data):
+    def decode(cls, data: BytesIO) -> int:
         value, i = 0, 0
         while True:
             b = data.read(1)
-            if not (b & 0x80):
+            if not (b & 0x80):  # type: ignore[operator]
                 break
-            value |= (b & 0x7F) << i
+            value |= (b & 0x7F) << i  # type: ignore[operator]
             i += 7
             if i > 63:
                 raise ValueError(f"Invalid value {value}")
-        value |= b << i
+        value |= b << i  # type: ignore[operator]
         return (value >> 1) ^ -(value & 1)
```

Comment:
Smth strange with types in this class, but I didn't find using VarInt64 in aiokafka, so I don't know what do with these errors.

Reply:
The problem here is that `data.read(1)` returns `bytes` and you need to unpack it the same way as in other places to get `int`. Another option might be to use `data.read(1)[0]`, but let's keep it consistent.

Reply:
Thanks, fixed